### DISCUSSION TITLE
Some questions about Waters MSE type data deconvolution

### DISCUSSION THREAD

**Discussion by gzzwjhzxh** [mobiusklein/ms_deisotope#24] [Q&A]
Hello developer，I am very pleased that you can provide a tool that supports deconvolution of data acquired in MSE mode. Can you give a brief introduction to the principle of using the deconvolution method? In the future, do you consider supporting exporting files for GNPS (see https://ccms-ucsd.github.io/GNPSDocumentation/featurebasedmolecularnetworking/), and whether you consider supporting mass spectrometry data with ion mobility.

**Comment by mobiusklein** [comment]
I assume this is w.r.t. the recent commit working on `MSEFrameGenerator` and the like? That's for a particular project dealing with data from a Waters Cyclic Ion Mobility QTOF-MS instrument. My intention for that is to make something that works for that project (MSE + IM), and have no plans beyond that for the moment.

The idea there is to treat each IM cycle like it is an LC-MS feature map, except replacing retention time with drift time. The algorithm then proceeds to extract raw m/z over drift time features, and uses a variation on the iterative dependency network-based deconvolution algorithm used for individual spectra to deconvolve those features in aggregate, yielding (neutral mass, charge) over drift time features. These IM-MS deconvolved features would be the product of the deconvolution algorithm, written out to mzML with each coordinate in the deconvolved features annotated with a "feature id" to make reconstruction simpler (though perhaps not space efficient).

Downstream from this would be an algorithm that reconstructs precursor-product relationships based upon overlapping elution profiles, which is the next phase of that project, and I've not gotten to the prototyping stage yet. In principle one could skip the IM extraction component if not using an IM instrument but still using MSE, but I don't have plans to deal with that situation.

Feature-based molecular networking is something on my reading list, and I'll see what that format looks like.

**Reply by gzzwjhzxh** [reply]
Thank you very much for your patient reply. The data I am currently getting is only the data collected in MSE mode. That is to say, currently ms_deisotope can be used for data analysis collected in MSE+IM mode,  however, since only the data collected by MSE has chromatographic peak overlap, the deconvoluted data may not be good.
